ws-server/src/server.rs

```diff
@@ -107,27 +108,28 @@ impl Server {
 							continue;
 						}
 
-						if Arc::strong_count(&conn_counter) > self.cfg.max_connections as usize {
+						if connections.len() > self.cfg.max_connections as usize {
 							log::warn!("Too many connections. Try again in a while");
 							continue;
 						}
+						id += 1;
+						let (tx, rx) = mpsc::channel::<()>(1);
+						connections.insert(id, rx);
 						let methods = methods.clone();
-						let counter = conn_counter.clone();
 						let cfg = self.cfg.clone();
 
 						tokio::spawn(async move {
-							let r = background_task(socket, id, methods, cfg).await;
-							drop(counter);
-							r
+							background_task(socket, id, methods, cfg, tx).await
+							// TODO: remove from connections.
```

Comment:
this is a little bit tricky, the straightforward way would be just to put the in connections in `Arc<Mutex<>>` however that won't work when it's cloned into the background thread without any further syncronization. TLDR, Arc::strong_count > 1 then drop has no effect.